### DISCUSSION TITLE
Airsim Collision Message to Drone or PC

### DISCUSSION THREAD

**Discussion by kwanyongcho7** [microsoft/AirSim#3480] [Support Q&A] [answered]
Hi, 

I am using px4 drone HITL mode in Airsim controlled by MAVSDK-Python.

I want to send message or notice when collision happens in Airsim to reboot or restart the flight using MAVSDK-python automatically.

Is there any way to let MAVSDK-Python know about collision happens? 

Thanks.

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @kwanyongcho7, you could maybe use the [AirSim python API](https://microsoft.github.io/AirSim/apis/) within your MAVSDK-python node to check for collision status through `simGetCollisionInfo()` and have that set a bool/flag or call a method within your node to restart the flight using MAVSDK-python?

**Reply by kwanyongcho7** [reply]
Is it also work with HITL? I am using my own FCC

**Reply by ahmed-elsaharti** [reply]
I think it would.

**Reply by kwanyongcho7** [reply]
Thanks it worked
